app/screens/thread/thread.tsx

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useMemo} from 'react';
+import {useIntl} from 'react-intl';
+import {BackHandler, StyleSheet, View} from 'react-native';
+import {Navigation} from 'react-native-navigation';
+import {Edge, SafeAreaView} from 'react-native-safe-area-context';
+
+import {fetchPostThread} from '@actions/remote/post';
+import {useServerUrl} from '@app/context/server';
+import {useTheme} from '@app/context/theme';
+import {changeOpacity} from '@app/utils/theme';
+import CompassIcon from '@components/compass_icon';
+import PostDraft from '@components/post_draft';
+import {General} from '@constants';
+import {THREAD_ACCESSORIES_CONTAINER_NATIVE_ID} from '@constants/post_draft';
+import {useAppState, useIsTablet} from '@hooks/device';
+import {dismissModal, mergeNavigationOptions} from '@screens/navigation';
+
+import ThreadPostList from './thread_post_list';
+
+import type ChannelModel from '@typings/database/models/servers/channel';
+import type PostModel from '@typings/database/models/servers/post';
+
+type ThreadProps = {
+    channel: ChannelModel;
+    componentId: string;
+    rootPost: PostModel;
+};
+
+const edges: Edge[] = ['left', 'right'];
+
+const getStyleSheet = StyleSheet.create(() => ({
+    flex: {
+        flex: 1,
+    },
+}));
+
+const CLOSE_BUTTON_ID = 'close-threads';
+
+const Thread = ({channel, componentId, rootPost}: ThreadProps) => {
+    const {formatMessage} = useIntl();
+    const appState = useAppState();
+    const isTablet = useIsTablet();
+    const serverUrl = useServerUrl();
+    const styles = getStyleSheet();
+    const theme = useTheme();
+
+    // Get post thread
+    useEffect(() => {
+        fetchPostThread(serverUrl, rootPost.id);
```

Comment:
I wonder if we can start fetching this earlier, even before showing the modal. I'm fine creating a function that calls this and shows the modal to reduce recurrent code

Reply:
Got it, Something like `switchToThread` 👍🏼 Working on it.